pkg/backend/aws/ecs.go

```diff
@@ -225,41 +327,67 @@ func (ab *Backend) getLogEventsForTask(
 		return errors.Wrap(err, "could not describe tasks")
 	}
 
-	// NOTE NOTE: we are making an assumption that we only have one container per task
-	//            this was here before, but I don't know if it is valid
-	container := tasksResult.Tasks[0].Containers[0]
-	if container.Reason != nil {
-		logrus.Warnf("container exited with status %s: %s", *container.LastStatus, *container.Reason)
+	if tasksResult == nil || len(tasksResult.Tasks) == 0 || len(*tasksResult.Tasks[0].TaskArn) == 0 {
+		return errors.Errorf("no matching tasks for task definition %s", taskDefARN)
 	}
-
-	// now the log group
-	logConfiguration := taskDefResult.TaskDefinition.ContainerDefinitions[0].LogConfiguration
-	logGroup, ok := logConfiguration.Options["awslogs-group"]
-	if !ok {
-		return errors.Errorf("could not infer log group")
+	taskId, err := ab.getTaskId(*tasksResult.Tasks[0].TaskArn)
+	if err != nil {
+		return errors.Wrap(err, "unable to determine a task id")
 	}
 
-	logPrefix, logPrefixSpecified := logConfiguration.Options["awslogs-stream-prefix"]
-	// Now we should have enough info to sort out the log stream
-	// see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html
-	// NOTE: this is REQUIRED for fargate, but shares logic with ECS as well
-	//       when prefix defined
-	taskARN := strings.Split(*container.TaskArn, "/")
-	logStream := taskARN[len(taskARN)-1]
-	if logPrefixSpecified {
-		// prefix-name/container-name/ecs-task-id
-		prefixName := logPrefix
-		containerName := taskDefResult.TaskDefinition.ContainerDefinitions[0].Name
-		ecsTaskID := logStream
-		logStream = path.Join(prefixName, *containerName, ecsTaskID)
+	logGroup, logStreamName, err := ab.getLogGroupAndStream(*taskDefResult.TaskDefinition, taskId, "")
+	if err != nil {
+		return errors.Wrap(err, "unable to determine log group and stream name")
 	}
 
-	return ab.getLogs(
+	return ab.GetLogs(
 		ctx,
-		&cwlv2.GetLogEventsInput{
+		&cloudwatchlogs.GetLogEventsInput{
 			LogGroupName:  &logGroup,
-			LogStreamName: &logStream,
+			LogStreamName: &logStreamName,
 		},
 		getlogs,
 	)
 }
+
+func (ab *Backend) getTaskId(taskArn string) (string, error) {
+	resourceArn, err := arn.Parse(taskArn)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to parse task ARN: '%s'", taskArn)
+	}
+
+	arnSegments := strings.Split(resourceArn.Resource, "/")
+	if len(arnSegments) < 3 {
+		return "", errors.Errorf("incomplete task ARN: '%s'", taskArn)
+	}
+	return arnSegments[len(arnSegments)-1], nil
+}
+
+func (ab *Backend) getLogGroupAndStream(taskDefinition ecstypes.TaskDefinition, taskId string, containerName string) (string, string, error) {
```

Comment:
```suggestion
func (ab *Backend) getLogGroupAndStreamNames(taskDefinition ecstypes.TaskDefinition, taskId string, containerName string) (string, string, error) {
```
what do you think of renaming like that? Otherwise, I thought that we would stream the log messages in this function as well